chain/src/chain.rs

```diff
@@ -279,7 +279,7 @@ impl ChainService {
     }
 
     // Truncate the main chain
-    // Use for testing only
+    // Use for testing only, can only truncate less than 50000 blocks each time
     pub(crate) fn truncate(&mut self, target_tip_hash: &Byte32) -> Result<(), Error> {
         let snapshot = Arc::clone(&self.shared.snapshot());
         assert!(snapshot.is_main_chain(target_tip_hash));
```

Comment:
I think we shouldn't `assert!` here, it may cause node panic.
If `target_tip_hash` is not in main chain, `fn truncate` should return `Err`.

Reply:
I think here is a double-check, custom error message will be given before calling `chain`:

https://github.com/nervosnetwork/ckb/blob/6de945f4512c2781fc17763cbabf1b3e9b580e08/rpc/src/module/test.rs#L497